src/test/java/org/kohsuke/github/GHContentIntegrationTest.java

```diff
@@ -217,6 +173,110 @@ public void testCRUDContent() throws Exception {
+                endsWith("/repos/hub4j-test-org/GHContentIntegrationTest/git/commits/" + gitCommit.getSHA1()));
+        assertThat(gitCommit.getNodeId(), notNullValue());
+        assertThat(gitCommit.getHtmlUrl().toString(),
+                equalTo("https://github.com/hub4j-test-org/GHContentIntegrationTest/commit/" + gitCommit.getSHA1()));
+        assertThat(gitCommit.getVerification(), notNullValue());
+
+        assertThat(ghCommit, notNullValue());
+        assertThat(ghCommit.getSHA1(), notNullValue());
+        assertThat(ghCommit.getUrl().toString(),
+                endsWith("/repos/hub4j-test-org/GHContentIntegrationTest/git/commits/" + ghCommit.getSHA1()));
+
+        return expectedRequestCount;
+    }
+
+    int checkCommitUserInfo(GitCommit gitCommit, GHCommit ghCommit, int expectedRequestCount)
+            throws IOException, ParseException {
+        assertThat(gitCommit.getAuthor().getName(), equalTo("Liam Newman"));
+        assertThat(gitCommit.getAuthor().getEmail(), equalTo("bitwiseman@gmail.com"));
+        assertThat(gitCommit.getCommitter().getName(), equalTo("Liam Newman"));
+        assertThat(gitCommit.getCommitter().getEmail(), equalTo("bitwiseman@gmail.com"));
+        assertThat("GHUser already resolved", mockGitHub.getRequestCount(), equalTo(expectedRequestCount));
+
+        assertThat(ghCommit.getAuthor().getName(), equalTo("Liam Newman"));
+        assertThat(ghCommit.getAuthor().getEmail(), equalTo("bitwiseman@gmail.com"));
+        assertThat(ghCommit.getCommitter().getName(), equalTo("Liam Newman"));
+        assertThat(ghCommit.getCommitter().getEmail(), equalTo("bitwiseman@gmail.com"));
+
+        return expectedRequestCount;
+    }
+
+    int checkCommitTree(GitCommit gitCommit, GHCommit ghCommit, int expectedRequestCount) throws IOException {
+        assertThat(gitCommit.getTreeSHA1(), notNullValue());
+        assertThat(gitCommit.getTreeUrl(),
+                endsWith("/repos/hub4j-test-org/GHContentIntegrationTest/git/trees/" + gitCommit.getTree().getSha()));
+        assertThat("GHTree already resolved", mockGitHub.getRequestCount(), equalTo(expectedRequestCount));
+
+        assertThat(ghCommit.getTree().getSha(), notNullValue());
+        assertThat("GHCommit has to resolve GHTree", mockGitHub.getRequestCount(), equalTo(expectedRequestCount += 1));
+        assertThat(ghCommit.getTree().getUrl().toString(),
+                endsWith("/repos/hub4j-test-org/GHContentIntegrationTest/git/trees/" + ghCommit.getTree().getSha()));
+        assertThat("GHCommit resolving GHTree is not cached",
+                mockGitHub.getRequestCount(),
+                equalTo(expectedRequestCount += 2));
+
+        return expectedRequestCount;
+    }
+
+    int checkCommitParents(GitCommit gitCommit, GHCommit ghCommit, int expectedRequestCount) throws IOException {
+        assertThat(gitCommit.getParentSHA1s(), hasSize(greaterThan(0)));
+        assertThat(ghCommit.getParentSHA1s(), hasSize(greaterThan(0)));
```

Comment:
Hm... maybe `hasSize()` calls something other than `size()`?  Try changing this test to 
```suggestion
        assertThat(gitCommit.getParentSHA1s().size(), is(greaterThan(0)));
        assertThat(ghCommit.getParentSHA1s().size(), is(greaterThan(0)));

        assertThat(gitCommit.getParentSHA1s().get(0), notNullValue());
        assertThat(ghCommit.getParentSHA1s().get(0), notNullValue());
```